### ISSUE TITLE
Improve runtime

### ISSUE THREAD

**Issue by constantinpape** [hci-unihd/batchlib#101]
The runtime has increased quite a bi in the last few days. In order to check new results a bit faster, it would be good if we can speed it up a bit again.

I can see the following reasons / what we can do about it:
- We run the analysis twice for IgG and IgA now. I don't think there's much we can do about this, apart from optimising the analysis code a bit.
- The background computation got a bit more involved, and it looks like this actually takes quite some time now too, but I haven't measured it properly.
- We still write the "summary images" in the `CellLevelAnalysis` job. I am not sure this is really necessary, if we can deactivate we can gain 2 x 15 minutes.

I can look into this once #94 is merged, but need to discuss with @imagirom first.

**Comment by imagirom** [comment]
> * The background computation got a bit more involved, and it looks like this actually takes quite some time now too, but I haven't measured it properly.

Yeah, the median over all bg pixel values on the cpu is the expensive part here.

**Comment by constantinpape** [comment]
Another thing I noted is that there are a couple of results that are always recomputed, even if everything should be there already. @imagirom have you set `force_recompute=True` somewhere?
Otherwise the validation is not working properly ...

**Comment by imagirom** [comment]
> @imagirom have you set `force_recompute=True` somewhere?

Not knowingly, I will check in `cell_analyis.py`.

@constantinpape Could you later post the runtime of the individual jobs?

**Comment by constantinpape** [comment]
> @constantinpape Could you later post the runtime of the individual jobs?

Yes, I will add this more explicitly to the logs and then post a log.

**Comment by constantinpape** [comment]
Here are the runtimes for one of the recent plates:
```
Preprocess: runtime: 1704.5831456184387 s
TorchPrediction: runtime: 377.65269565582275 s
StardistPrediction: runtime: 2308.3338465690613 s
SeededWatershed: runtime: 187.56252217292786 s
InstanceFeatureExtraction: runtime: 1267.5676238536835 s
ImageLevelQC: runtime: 71.2727723121643 s
ExtractBackground: runtime: 321.7103023529053 s
FindInfectedCells: runtime: 81.8150155544281 s
CellLevelQCserum_IgG_corrected: runtime: 53.93965196609497 s
ImageLevelQCserum_IgG_corrected: runtime: 30.466031312942505 s
WellLevelQCserum_IgG_corrected: runtime: 65.1491870880127 s
CellLevelAnalysisserum_IgG_corrected: runtime: 156.49572443962097 s
CellLevelQCserum_IgA_corrected: runtime: 45.548330783843994 s
ImageLevelQCserum_IgA_corrected: runtime: 30.426975965499878 s
WellLevelQCserum_IgA_corrected: runtime: 54.87663388252258 s
CellLevelAnalysisserum_IgA_corrected: runtime: 161.9966504573822 s
MergeAnalysisTables: runtime: 0.8329212665557861 s
DbResultWriter: runtime: 1.9223129749298096 s
SlackSummaryWriter: runtime: 24.24576210975647 s

CellAnalysisWorkflow in 7899.495049238205s
```

**Comment by constantinpape** [comment]
So stardist is still the clear bottleneck.
